app/src/test/kotlin/compiler/e2e/NameResolutionErrorsTest.kt

```diff
@@ -1,5 +1,246 @@
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.NameConflict::class
+        )
+    }
+
+    @Ignore
+    @Test
+    fun `test conflicts (functions)`() {
+        assertErrorOfType(
+            """
+                    czynność f() {
+                        czynność g() -> Liczba {
+                            zwróć 17
+                        }
+                        czynność g() -> Liczba {
+                            zwróć 18
+                        }
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.NameConflict::class
+        )
+    }
+
+    @Ignore
+    @Test
+    fun `test conflicts (functions with different return types)`() {
+        assertErrorOfType(
+            """
+                    czynność f() -> Liczba {
+                        czynność g() -> Czy {
+                            zwróć prawda
+                        }
+                        czynność g() -> Nic {
+                            zakończ
+                        }
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.NameConflict::class
+        )
+    }
+
+    @Ignore
+    @Test
+    fun `test conflicts (functions with different signatures)`() {
+        assertErrorOfType(
+            """
+                    czynność f() -> Liczba {
+                        czynność g(a: Liczba) { }
+                        czynność g(a: Czy) { }
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.NameConflict::class
+        )
+    }
+
+    // Assuming no functional features.
+
+    @Ignore
+    @Test
+    fun `test calling variables`() {
+        assertErrorOfType(
+            """
+                    czynność f() -> Czy {
+                        zm x: Czy = prawda
+                        zwróć x()
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.VariableIsNotCallable::class
+        )
+    }
+
+    // I'm not 100% sure if the next two tests make sense
+    @Ignore
+    @Test
+    fun `test using functions in assignment`() {
+        assertErrorOfType(
+            """
+                    czynność f() {
+                        czynność g() { }
+                        zm x: Liczba = x
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.FunctionIsNotVariable::class
+        )
+    }
+
+    @Ignore
+    @Test
+    fun `test using functions in conditions`() {
+        assertErrorOfType(
+            """
+                    czynność f() -> Czy {
+                        czynność g() { }
+                        zwróć (g == 17)
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.FunctionIsNotVariable::class
+        )
+    }
```

Comment:
Could you please add some tests for function parameters? We could test, for example, name conflicts (between two parameters and between a parameter and a variable), treating parameters as callables, using parameters outside function scope, and maybe using undefined variables/functions as default parameter values:
```
czynność f(x: Liczba = y) {}
czynność f(x: Liczba = f()) {}
```